git-config/src/file/git_config.rs

```diff
@@ -145,28 +172,70 @@ impl<'event> GitConfig<'event> {
     ///
     /// [`git-config`'s documentation]: https://git-scm.com/docs/git-config#Documentation/git-config.txt-FILES
     #[inline]
-    pub fn from_paths(paths: &[&Path]) -> Result<Self, ParserOrIoError<'static>> {
+    pub fn from_paths(paths: &[&Path], git_install_dir: Option<&std::path::Path>) -> Result<Self, from_paths::Error> {
         let mut config = Self::new();
-
-        for path in paths {
-            let other = Self::open(path)?;
-            for (section_id, section_header) in other.section_headers {
-                config.push_section(
-                    section_header.name.0.to_owned(),
-                    section_header.subsection_name.to_owned(),
-                    other.sections[&section_id].clone(),
-                );
+        let mut seen = HashSet::new();
+        let paths: Vec<_> = paths.iter().map(|p| p.into()).collect();
+        const MAX_DEPTH: u8 = 10;
```

Comment:
Since this is a plumbing crate, I think it should be the application that decides where this limit is. Could this be a parameter?

If so, this might pose the question of these parameters, all related to resolving includes, should be part of an `Options` struct of sorts.

Reply:
like  `pub struct Options<'a> {` ?

Reply:
That looks good to me!